File: Sources/AdbShell/Models/AppInfo.swift
import Foundation

struct InstalledApp: Identifiable, Hashable {
    var id: String { packageName }
    let packageName: String
    var isSystem: Bool
    var isEnabled: Bool
}

struct AppPermission: Identifiable, Hashable {
    var id: String { name }
    let name: String
    var granted: Bool
    var isRuntime: Bool

    var shortName: String {
        if let last = name.split(separator: ".").last {
            return String(last)
        }
        return name
    }
}

struct AppDetail {
    let packageName: String
    var versionName: String?
    var versionCode: String?
    var firstInstallTime: String?
    var lastUpdateTime: String?
    var targetSdk: String?
    var apkPath: String?
    var isEnabled: Bool
    var permissions: [AppPermission]
}

File: desktop/src/main/adb/types/AppInfo.ts
// Порт Sources/AdbShell/Models/AppInfo.swift

export interface InstalledApp {
  packageName: string;
  isSystem: boolean;
  isEnabled: boolean;
}

export interface AppPermission {
  name: string;
  granted: boolean;
  isRuntime: boolean;
}

export function permissionShortName(permission: AppPermission): string {
  const parts = permission.name.split('.');
  return parts.length > 0 ? parts[parts.length - 1] : permission.name;
}

export interface AppDetail {
  packageName: string;
  versionName?: string;
  versionCode?: string;
  firstInstallTime?: string;
  lastUpdateTime?: string;
  targetSdk?: string;
  apkPath?: string;
  isEnabled: boolean;
  permissions: AppPermission[];
  /** Linux UID приложения на устройстве (userId= в dumpsys package). */
  uid?: number;
}
